src/openrct2/world/sprite.c

```diff
@@ -285,6 +287,7 @@ static void sprite_reset(rct_unk_sprite *sprite)
 
     sprite->linked_list_type_offset = llto;
     sprite->next = next;
+    sprite->next_in_quadrant = next_in_quadrant;
```

Comment:
Can you also add `sprite->sprite_identifier = SPRITE_IDENTIFIER_NULL;` to this function and change `sprite_clear_all_unused` to use it?